addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1222,7 +1222,7 @@ def _notify_thread_by_web_push(self, message, recipients_data, msg_vals=False, *
                  'type': 'customer',
                  'groups': [],
                  }
-                for partner in chat_channels.mapped("channel_partner_ids")
+                for partner in chat_channels.mapped("channel_partner_ids") if not self.channel_member_ids.filtered(lambda member: member.partner_id == partner).mute_until_dt
```

Comment:
This condition doesn't look correct, it should iterate channel members directly instead of channel partners.

Reply:
why? The previous code shows only partners are added to the `recipients_data`

Reply:
The line is making a double loop on `channel_partner_ids` / `channel_member_ids`.

-> `chat_channels.channel_member_ids.filtered(lambda m: not m.mute_until_dt).partner_id`